motioneye/update.py

```diff
@@ -34,17 +34,23 @@ def get_os_version():
 
 
 def _get_os_version_lsb_release():
-    try:
-        output = subprocess.check_output('lsb_release -sri', shell=True)
-        lines = output.strip().split()
-        name, version = lines
-        if version.lower() == 'rolling':
-            version = ''
+    # check if it exists before calling to avoid log spam
+    rc = subprocess.call(['which', 'lsb_release'])
+
+    if rc == 0:
+        try:
+            output = subprocess.check_output('lsb_release -sri', shell=True)
+            lines = output.strip().split()
+            name, version = lines
+            if version.lower() == 'rolling':
+                version = ''
         
-        return name, version
+            return name, version
 
-    except:
-        return _get_os_version_uname()
+        except:
+            return _get_os_version_uname()
+    else:
+            return _get_os_version_uname()
```

Comment:
Wrong indentation? Simpler may be:
```py
    # check if it exists before calling to avoid log spam
    if subprocess.call(['which', 'lsb_release']):
        try:
            output = subprocess.check_output('lsb_release -sri', shell=True)
            lines = output.strip().split()
            name, version = lines
            if version.lower() == 'rolling':
                version = ''

            return name, version

        except:
            pass

    return _get_os_version_uname()
```

Reply:
Can we lose please the bare except as discussed in PEP8 and
https://realpython.com/the-most-diabolical-python-antipattern

Reply:
I also see annotations on this in CodeFactor. What is the exception type in case of a failing `subprocess.check_output`?

Reply:
`subprocess.CalledProcessError` it seems. So:
```py
except subprocess.CalledProcessError as e:
    log.error('Running lsb_release failed with: %s' % e.output)
```
Or something like that?